### PR TITLE
Extending some return types in annotation docblocks

### PR DESCRIPTION
**Pull request by Theaxiom**
This is me splitting up a larger PR, please see #10532 for details.

### CODE REVIEW

File: src/Console/ConsoleOptionParser.php [hunk 1]
```diff
@@ -298,7 +298,7 @@ public function getCommand()
      *
      * @deprecated 3.4.0 Use setCommand()/getCommand() instead.
      * @param string|null $text The text to set, or null if you want to read
-     * @return string|$this If reading, the value of the command. If setting $this will be returned.
+     * @return ConsoleOptionParser|string If reading, the value of the command. If setting $this will be returned.
```
Comment (dereuromark) [inline src/Console/ConsoleOptionParser.php]: Please use FQCN in doc blocks: \...\

Comment (dereuromark) [inline src/Console/ConsoleOptionParser.php]: In these cases $this is correct though afaik - the IDE knows that this is chaining and knows the type implicitly.

**Comment by dereuromark** [comment]
👎 as chaining demands this kind of doc block syntax according to phpdocs.

File: src/Console/ConsoleOptionParser.php (hunk 1 above)
Comment (Theaxiom) [inline src/Console/ConsoleOptionParser.php]: Alright, I will update them to FQCN's.

**Comment by ADmad** [comment]
Please don't change any usage of `$this` in param types. It's usage has been decided through enough discussion in the past based on what IDE's expect.

**Pull request closed without merging**